pandas/core/indexes/interval.py

```diff
@@ -124,7 +124,11 @@ def setop_check(method):
     def wrapped(self, other, sort=False):
         self._validate_sort_keyword(sort)
         self._assert_can_do_setop(other)
-        other = ensure_index(other)
+        other, _ = self._convert_can_do_setop(other)
+
+        if op_name == "intersection":
```

Comment:
why is this *only* the case?

Reply:
bc there's a lot of cases to address and im handling them one at a time